### PR TITLE
Some improvements

### PR DESCRIPTION
**Pull request by martijncalker**

### CODE REVIEW

File: lib/Providers/Rng/RNGException.php [hunk 1]
```diff
@@ -2,10 +2,6 @@
...
-{
-    function __construct($message = "", $code = 0, $exception = null)
-    {
-    	parent::__construct($message, $code, $exception);
-    }
-}
\ No newline at end of file
+use RobThree\Auth\TwoFactorAuthException;
+
+class RNGException extends TwoFactorAuthException {}
```
Comment (martijncalker) [inline lib/Providers/Rng/RNGException.php]: I have changed this to `TwoFactorAuthException`, is that OK for you? I can do the other as well (I forgot it I see)

**Comment by martijncalker** [comment]
I did typehint the arguments in the constructor; I could change the tests to `@expectedException TypeError` (if you still want to test this)

**Comment by RobThree** [comment]
As it currently stands the changes break the test(s). I haven't look at it closely yet (too busy, later this week) but I don't want to break compatibility etc. so that will have to be checked.

**Comment by martijncalker** [comment]
As I understand the code you're checking for `instanceof Providers\Qr\IQRCodeProvider ` and throw an exception if not. By typehint the argument in the constructor you don't have to check it yourself (php will throw the TypeError) and with the `= null`you can still leave it 'empty'/null.

**Comment by RobThree** [comment]
I know but I'm not sure if all the old(er) PHP versions support typehinting? Currently 5.3 and up is supported. I could be mistaken but from the top of my head isn't typehinting something that was introduced in 5.4 or later?

**Comment by martijncalker** [comment]
The [docs](http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration) and web are not very clear about that...

You can also tag this repo ~ v2 for this change (and add the PSR recommendations also in it)

**Comment by martijncalker** [comment]
I guess It wouldn't break things ;-)

How do you want me to fixes the tests? I personally prefer to remove the tests for throwing an exception when constructing the TwoFactorAuth with wrong arguments provided. 

And whats's your opinion about the `RNGException extends TwoFactorAuthException`? Should I remove it or extend the `QRException` as well?

**Comment by RobThree** [comment]
> How do you want me to fixes the tests? I personally prefer to remove the tests for throwing an exception when constructing the TwoFactorAuth with wrong arguments provided.

If I'm not mistaken this is [an issue with PHPUnit](https://github.com/sebastianbergmann/phpunit/issues/1748). Assuming there's no decent workaround until implemented by PHPUnit I guess removing the tests is the only way to go.

> And whats's your opinion about the `RNGException extends TwoFactorAuthException`? Should I remove it or extend the `QRException` as well?

Extending `TwoFactorAuthException` actually makes sense (and I wonder why I didn't come up with that in the first place).

Other than that your changes nicely clean up the code, improve the quality and readability and seem to be fine to me! Thank you very much!

**Comment by martijncalker** [comment]
No problem, I think it's ready now :)

**Comment by RobThree** [comment]
Et voila! 👍 
Thanks @martijncalker 🎉

**Comment by RobThree** [comment]
WTH... somehow the merge didn't happen. I'll redo the PR soon.

**Pull request closed without merging**